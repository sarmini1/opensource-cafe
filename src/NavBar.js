import React from "react";
import "./NavBar.css";
import { NavLink } from "react-router-dom";
import { Navbar, Nav, NavItem } from "reactstrap";

/** NavBar
 * 
 * Props:
 * - none
 * 
 * State:
 * - none
 * 
 * App --> NavBar
 */
function NavBar() {
  return (
    <div>
      <Navbar expand="md">
        <NavLink exact to="/" className="navbar-brand">
          openSource Cafe
        </NavLink>

        <Nav className="ml-auto" navbar>
          <NavItem>
            <NavLink className="NavLink-snacks" to="/snacks">Snacks</NavLink>
          </NavItem>
          <NavItem>
            <NavLink className="NavLink-drinks" to="/drinks">Drinks</NavLink>
          </NavItem>
          <NavItem>
            <NavLink className="NavLink-new-item" to="/add">Add an Item!</NavLink>
          </NavItem>
        </Nav>
      </Navbar>
    </div>
  );
}

export default NavBar;