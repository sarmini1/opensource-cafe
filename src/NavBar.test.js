import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { MemoryRouter } from "react-router-dom";
import NavBar from "./NavBar";

it('mounts without crashing', function () {
  render(<MemoryRouter><NavBar /></MemoryRouter>);
});

it("matches snapshot", function () {
  const { container } = render(<MemoryRouter><NavBar /></MemoryRouter>);
  expect(container).toMatchSnapshot();
});

it('displays appropriate links', function () {
  const { getByText } = render(<MemoryRouter><NavBar /></MemoryRouter>);

  const homeLink = getByText(/openSource Cafe/i);
  expect(homeLink).toBeInTheDocument();

  const snackLink = getByText(/Snacks/i);
  expect(snackLink).toBeInTheDocument();

  const drinkLink = getByText(/Drinks/i);
  expect(drinkLink).toBeInTheDocument();

  const newItemFormLink = getByText(/Add an Item!/i);
  expect(newItemFormLink).toBeInTheDocument();
});

it('navigates to links correctly', function () {
  const { container } = render(<MemoryRouter><NavBar /></MemoryRouter>);

  const snackLink = container.querySelector('.NavLink-snacks');
  fireEvent.click(snackLink);
});