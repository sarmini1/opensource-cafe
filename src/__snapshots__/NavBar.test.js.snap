// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="NavBar"
  >
    <nav
      class="navbar navbar-expand-md"
    >
      <div
        class="container-fluid"
      >
        <a
          aria-current="page"
          class="navbar-brand active"
          href="/"
        >
          openSource Cafe
        </a>
        <ul
          class="ml-auto navbar-nav"
        >
          <li
            class="nav-item"
          >
            <a
              class="NavLink-snacks"
              href="/snacks"
            >
              Snacks
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="NavLink-drinks"
              href="/drinks"
            >
              Drinks
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="NavLink-new-item"
              href="/add"
            >
              Add an Item!
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</div>
`;
